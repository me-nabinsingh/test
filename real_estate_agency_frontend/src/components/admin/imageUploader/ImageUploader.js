import React, { Component } from 'react';
import { FilePond, File, registerPlugin } from 'react-filepond';
import 'filepond/dist/filepond.min.css';
import FilePondPluginImageExifOrientation from 'filepond-plugin-image-exif-orientation';
import FilePondPluginImagePreview from 'filepond-plugin-image-preview';
import 'filepond-plugin-image-preview/dist/filepond-plugin-image-preview.css';
import config from '../../../config';

// Register the plugins
registerPlugin(FilePondPluginImageExifOrientation, FilePondPluginImagePreview);

export default class  ImageUploader extends Component {
    state = {
        files: []
    }

    render() {
        
        return (
            <FilePond ref={ref => this.pond = ref}
                    allowMultiple={true} 
                    server= {`${config.apiUrl}/images`}
                    onload = {(response) => {
                        console.log(response);
                    }}
                    onupdatefiles={(fileItems) => {
                        // Set current file objects to this.state
                        this.setState({
                            files: fileItems.map(fileItem => {
                                return fileItem.file;
                            })
                        });
                    }}>
            
                {this.state.files.map(file => (
                    <File key={file} src={file} origin="local" />
                ))}
                
            </FilePond>
        );
    }
}
 
