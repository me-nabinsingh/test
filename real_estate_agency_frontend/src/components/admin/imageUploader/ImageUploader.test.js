import React from 'react';
import ImageUploader from './ImageUploader';
import renderer from 'react-test-renderer';

describe('<ImageUploader />', () => {

    it('renders without crashing', () => {
        const component = renderer.create(
            <ImageUploader/>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});