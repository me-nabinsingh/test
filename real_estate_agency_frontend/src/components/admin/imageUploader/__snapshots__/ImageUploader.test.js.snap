// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageUploader /> renders without crashing 1`] = `
<div
  className="filepond--wrapper"
>
  <input
    multiple={true}
    type="file"
  />
</div>
`;
